globals.go

```diff
@@ -276,20 +279,22 @@ var InternalAliases []string = []string{
 	GitTagAlias,
 }
 
-var TaskFailureStatuses []string = []string{
+var TaskSystemFailureStatuses []string = []string{
 	TaskTimedOut,
-	TaskFailed,
 	TaskSystemFailed,
 	TaskTestTimedOut,
 	TaskSetupFailed,
 	TaskSystemUnresponse,
 	TaskSystemTimedOut,
 }
 
+var TaskFailureStatuses []string = append([]string{TaskFailed}, TaskSystemFailureStatuses...)
```

Comment:
Documentation.

Reply:
I would avoid doing appending in that order because [append can mutate the array underlying the slice](https://go.dev/ref/spec#Appending_and_copying_slices).

> If the capacity of s is not large enough to fit the additional values, append allocates a new, sufficiently large underlying array that fits both the existing slice elements and the additional values. Otherwise, append re-uses the underlying array.

Reply:
I feel like that isn't super relevant here since I assume we know the underlying array size of TaskSystemFailureStatuses because it's a constant; I am mostly biased because this looks ugly and someone will likely change it in the future anyway (so if we really don't want to change it we should add a comment to explain, although I think it would be fine to change)

Reply:
The issue is mostly if you ever try to append to `TaskSystemFailureStatuses` in another place, it may mutate the underlying array and produce unexpected results for those who are appending to it.
```go
import "fmt"

func main() {
	s0 := make([]string, 0, 3)
	s0 = append(s0, "one", "two")
	s1 := append(s0, "three")
	s2 := append(s0, "four")
	fmt.Println(s0) // [one two]
	fmt.Println(s1) // [one two four]
	fmt.Println(s2) // [one two four]
}

```

Reply:
😭  that's devastating. Thanks for the example.